File: leetcode/0350-intersection-of-two-arrays-ii/solution.py
class Solution(object):
    def intersect(self, nums1, nums2):
        d={}
        nums=[]
        for i in nums1:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        for j in nums2:
            if j in d and d[j]!=0:
                d[j]-=1
                nums.append(j)
        return nums
            
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        

File: leetcode/0350-intersection-of-two-arrays-ii/solution.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        if (nums1.size()>nums2.size()){
            for (int i: nums2){
                auto it=find(nums1.begin(),nums1.end(),i);
                if (it!=nums1.end()){
                    res.push_back(i);
                    nums1.erase(it);
                }
            }
            return res;
        }
        for (int i: nums1){
                auto it=find(nums2.begin(),nums2.end(),i);
                if (it!=nums2.end()){
                    res.push_back(i);
                    nums2.erase(it);
                }
            }
            return res;

    }
};
